config/rbac/leader_election_role.yaml

```diff
@@ -0,0 +1,37 @@
+# permissions to do leader election.
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: leader-election-role
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - configmaps
+    verbs:
+      - get
+      - list
+      - watch
+      - create
+      - update
+      - patch
+      - delete
```

Comment:
Why are full config map permissions required in the namespace?

Reply:
This role is auto-generated by kubebuilder. I am not sure if all these permissions are really required during leader-election or not. Will check and update here

Reply:
Alright I just tested and looks like it needs all the permissions on cm except for delete